db.go

```diff
@@ -397,6 +393,19 @@ func (db *DB) close() error {
 	return nil
 }
 
+func (db *DB) ensureFreelist() *freelist {
```

Comment:
Sometimes you want to have single code for both readonly open and readwrite simply cause you don't know will you write or not.

Anyway, I was never confident last commit will be accepted.